kart/base_diff_writer.py

```diff
@@ -256,9 +257,25 @@ def write_warnings_footer(self):
                 err=True,
             )
 
-    def write_diff(self):
+    def write_diff(self, diff_format=DiffFormat.FULL):
         """Default implementation for writing a diff. Subclasses can override."""
         self.write_header()
+        # If the diff format is NONE, skip getting the diff
+        if diff_format == DiffFormat.NONE:
+            self.write_warnings_footer()
+            return
+
+        # If the diff format is NODATACHANGES, check if there is a diff and print True or False
+        elif diff_format == DiffFormat.NODATACHANGES:
+            for ds_path in self.all_ds_paths:
+                ds_diff = self.get_dataset_diff(ds_path)
+                has_changes = bool(ds_diff)
+            if has_changes:
+                click.secho(f"    Data changes: {has_changes}\n")
+            self.write_warnings_footer()
```

Comment:
there's something wrong here. Things to think about:

1. what happens when there is a meta-only diff (no feature changes)? Perhaps we need a test to make sure this does the right thing in that case.
2. what happens when there are multiple datasets?
3. do any subclasses of `BaseDiffWriter` call this base `write_diff` implementation? Is the output appropriate for BaseDiffWriter, or should this code move to a subclass? (we might need to discuss this one on a call)

Reply:
(This part of the change has been reverted / reworked, so these comments have been addressed)